lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -210,38 +255,48 @@ class Parameter:
     Note that a parameter that's "required" must also be non-empty.
     """
 
-    def __init__(self, name: AnyStr, type: ParamType, required: bool = False):
+    def __init__(
+        self,
+        name: AnyStr,
+        type: ParamType,
+        required: bool = False,
+        nullable: bool = False,
+    ):
         """
-        __init__ Initialize a Parameter object describing a JSON parameter
-        with its type and attributes.
+        Initialize a Parameter object describing a JSON parameter with its type
+        and attributes.
 
         Args:
             name: Parameter name
             type: Parameter type
             required: whether the parameter is required (default to False)
+            nullable: whether the parameter value can be null
         """
         self.name = name
         self.type = type
         self.required = required
+        self.nullable = nullable
 
     def invalid(self, json: Dict[AnyStr, Any]) -> bool:
         """
-        Check whether the value of this parameter in the JSON document
-        is invalid.
-
-        If the parameter is required and missing (or null), then it's invalid;
-        if the parameter is not required and specified as null, it's invalid.
+         Check whether the value of this parameter in the JSON document
+         is invalid.
 
         Args:
-            json: The client JSON document being validated.
+             json: The client JSON document being validated.
 
-        Returns:
-            True if unacceptable
+         Returns:
+             True if the specified value is unacceptable
         """
-        return not json[self.name] if self.name in json else self.required
+        if self.name in json:
+            if json[self.name] is None or json[self.name] == "":
+                return not self.nullable
+            return False
+        else:
+            return self.required
 
     def __str__(self) -> str:
-        return f"Parameter<{self.name}:{self.type}>"
+        return f"Parameter<{self.name}:{self.type}(Req:{self.required},Null:{self.nullable})>"
```

Comment:
I kinda like the explicit confirmation that it's falsey instead of implying that by the keyword's omission, which can lead to checking the code to be sure what's being reported. But there's always `repr` and you have a point that making `__str__` simpler and more conversational is probably appropriate. (Although I think I'd stick with "nullable" rather than the somewhat awkward "null-OK".)